cmd/root.go

```diff
@@ -120,12 +119,19 @@ func initConfig() {
 		viper.SetConfigFile(cfgFile)
 	}
 
-	viper.SetConfigName(".checks") // name of config file (without extension)
-	viper.AddConfigPath("$HOME")   // adding home directory as first search path
-	viper.AutomaticEnv()           // read in environment variables that match
+	viper.SetConfigName("dcos-checks-config") // name of config file (without extension)
+	viper.AddConfigPath("/opt/mesosphere/etc/")
+	viper.AutomaticEnv()
 
 	// If a config file is found, read it in.
 	if err := viper.ReadInConfig(); err == nil {
 		logrus.Infof("Using config file: %s", viper.ConfigFileUsed())
+		DCOSConfig.Role = viper.GetString("role")
```

Comment:
Should we prefer to change the condition here to `err != nil` and then fail and early in that clause, and then set the DSOSConfig properties after the if conditional?